utils/src/server/rpc_module.rs

```diff
@@ -252,6 +260,51 @@ impl SubscriptionSink {
 		for entry in errored {
 			subs.remove(&entry);
 		}
+
 		Ok(())
 	}
+
+	/// Send a message to all subscriptions that could parse `P` as input.
+	///
+	/// F: is a closure that you need to provide to apply on the input P.
+	pub fn send_each<T, F>(&self, f: F) -> anyhow::Result<()>
+	where
+		F: Fn(&mut Params) -> T,
```

Comment:
`Params` being `&mut` look like a foot gun to me.

I also think the closure should support error handling in user land:

`F: Fn(&Params) -> anyhow::Result<Option<T>>`, where:

+ `Ok(Some(T))` - no errors, send `T` to the subscriber.
+ `Ok(None)` - no errors, don't send anything to the subscriber.
+ `Err(_)` - error, log it (not sure if we need to send anything back?).

Reply:
It would have to be `F: Fn(&Option<Params>) -> anyhow::Result<Option<T>>` I think?

This __seems__ to work:

```rust
	pub fn send_each<T, F>(&self, f: F) -> anyhow::Result<()>
	where
		F: Fn(&Option<Params>) -> anyhow::Result<Option<T>>,
		T: Serialize,
	{
		let mut subs = self.subscribers.lock();
		let mut errored = Vec::new();

		for ((conn_id, sub_id), sender) in subs.iter_mut() {
			match f(&sender.params) {
				Ok(Some(res)) => {
					let msg = serde_json::to_string(&JsonRpcSubscriptionResponse {
						jsonrpc: TwoPointZero,
						method: self.method,
						params: JsonRpcNotificationParams { subscription: *sub_id, result: &to_raw_value(&res).expect("TODO what to do here?") },
					})?;

					// Mark broken connections, to be removed.
					if sender.sink.unbounded_send(msg).is_err() {
						errored.push((*conn_id, *sub_id));
					}
				},
				Ok(None) => log::warn!("Calling `send_each` for subscriptions that don't use params is… bad?"),
				Err(e) => log::error!("err: {:?}", e)
			}
		}

		// Remove broken connections
		for entry in errored {
			subs.remove(&entry);
		}

		Ok(())
	}
```

Reply:
Params being `Option<Params>` instead of just `Params` is a separate issue, see [the other comment](https://github.com/paritytech/jsonrpsee/pull/318#discussion_r635927172).

Reply:
yeah, that's neat.